tests/generic_config_updater/test_pfcwd_status.py

```diff
@@ -0,0 +1,237 @@
+    """
+    Tests GCU config for pfcwd stop scenario
+        1. Covers the case for stopping pfcwd on single port and all ports
+        2. Verifies that the config is as expected in CONFIG DB
+        3. Validates the number of PFC_WD related entries in FLEX DB is as expected
+        4. Validates that orchagent is running fine pre and post test
+    """
+    pfcwd_config = extract_pfcwd_config
+
+    if port == 'single':
+        expected_count = get_flex_db_count(duthost) - 3
+    else:
+        expected_count = 0
+    json_patch = list()
+    exp_str = 'Ethernet'
+    for interface in pfcwd_config:
+        json_patch.extend([
+                            {
+                              'op': 'remove',
+                              'path': '/PFC_WD/{}'.format(interface)
+                            }])
+        if port == 'single':
+            exp_str = interface
+            break
+
+    try:
+        tmpfile = generate_tmpfile(duthost)
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+        pfcwd_updated_config = duthost.shell("show pfcwd config")
+        pytest_assert(not pfcwd_updated_config['rc'], "Unable to read updated pfcwd config")
+        pytest_assert(exp_str not in pfcwd_updated_config['stdout'].split(),
+                      "pfcwd unexpectedly still running")
+        check_config_update(duthost, expected_count)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+@pytest.mark.parametrize('port', ['single', 'all'])
+def test_start_pfcwd(duthost, extract_pfcwd_config, ensure_dut_readiness, stop_pfcwd, port):
+    """
+    Tests GCU config for pfcwd start scenario
+        1. Covers the case for starting pfcwd on single port and all ports
+        2. Verifies that the config is as expected in CONFIG DB
+        3. Validates the number of PFC_WD related entries in FLEX DB is as expected
+        4. Validates that orchagent is running fine pre and post test
+    """
+    pfcwd_config = extract_pfcwd_config
+
+    if port == 'single':
+        expected_count = 3
+    else:
+        expected_count = len(pfcwd_config) * 3
+    json_patch = list()
+    exp_str = 'Ethernet'
+    for interface, value in pfcwd_config.items():
+        json_patch.extend([
+                            {
+                              'op': 'add',
+                              'path': '/PFC_WD/{}'.format(interface),
+                              'value': {'action': value['action'],
+                                        'detection_time': value['detect_time'],
+                                        'restoration_time': value['restore_time']}}])
+        if port == 'single':
+            exp_str = interface
+            break
+
+    try:
+        tmpfile = generate_tmpfile(duthost)
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
```

Comment:
I have opened a PR to check for if GCU is being used on an allowed platform based on RDMA support: https://github.com/sonic-net/sonic-mgmt/pull/8341/files
 
Request to wait until PR #8341 is merged to use the helper function: 
https://github.com/sonic-net/sonic-mgmt/blob/75031e4fc1cb60913787d485ae338a0c5a57465c/tests/generic_config_updater/gu_utils.py#L328

so that we only expect_op_success on allowed platforms.

Reply:
Thanks @isabelmsft for the heads up. I have a comment on the other PR.  Based on your feedback there, I will make the change for this testcase

Reply:
updated